extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -249,32 +273,70 @@ void PerformanceDataMonitor::addCustomPDHCountersFromProperty(const std::string&
   }
 }
 
-void PerformanceDataMonitor::setupMembersFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
-  std::string predefined_groups;
-  if (context->getProperty(PredefinedGroups.getName(), predefined_groups)) {
-    logger_->log_trace("Predefined group configured to be %s", predefined_groups);
-    addCountersFromPredefinedGroupsProperty(predefined_groups);
-  }
-
+void PerformanceDataMonitor::setupCountersFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
   std::string custom_pdh_counters;
   if (context->getProperty(CustomPDHCounters.getName(), custom_pdh_counters)) {
     logger_->log_trace("Custom PDH counters configured to be %s", custom_pdh_counters);
     addCustomPDHCountersFromProperty(custom_pdh_counters);
   }
+}
+
+void PerformanceDataMonitor::setupPredefinedGroupsFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string predefined_groups;
+  if (context->getProperty(PredefinedGroups.getName(), predefined_groups)) {
+    logger_->log_trace("Predefined group configured to be %s", predefined_groups);
+    addCountersFromPredefinedGroupsProperty(predefined_groups);
+  }
+}
 
+void PerformanceDataMonitor::setupOutputFormatFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
   std::string output_format_string;
   if (context->getProperty(OutputFormatProperty.getName(), output_format_string)) {
     if (output_format_string == OPEN_TELEMETRY_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be OpenTelemetry");
       output_format_ = OutputFormat::OPENTELEMETRY;
     } else if (output_format_string == JSON_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be JSON");
       output_format_ = OutputFormat::JSON;
     } else {
-      logger_->log_error("Invalid OutputFormat, defaulting to JSON");
-      output_format_ = OutputFormat::JSON;
+      throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Invalid PerformanceDataMonitor Output Format: " + output_format_string);
     }
   }
+
+  std::string output_compactness_string;
+  if (context->getProperty(OutputCompactness.getName(), output_compactness_string)) {
+    if (output_compactness_string == PRETTY_FORMAT_STR) {
+      pretty_output_ = true;
+    } else if (output_compactness_string == COMPACT_FORMAT_STR) {
+      pretty_output_ = false;
+    } else {
+      throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Invalid PerformanceDataMonitor Output Compactness: " + output_compactness_string);
+    }
+  }
+
+  logger_->log_trace("OutputFormat is configured to be %s %s", pretty_output_ ? "pretty" : "compact", output_format_ == OutputFormat::JSON ? "JSON" : "OpenTelemetry");
+}
+
+void PerformanceDataMonitor::setupDecimalPlacesFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string decimal_places_str;
+  if (context->getProperty(DecimalPlaces.getName(), decimal_places_str) && decimal_places_str == "") {
```

Comment:
I think this should be
```suggestion
  if (!context->getProperty(DecimalPlaces.getName(), decimal_places_str) || decimal_places_str == "") {
```
although I don't know if `getProperty()` can return false when there is a default.

Reply:
I don't think it can, what are the benefits of using the empty string instead of leaving it without a default?